### ISSUE TITLE
Travis CI failing: Can't open hdf5 file

### ISSUE THREAD

**Issue by MichaelCurrie** [openworm/movement_validation#147]
Travis CI seems to be able to open

- `example_video_norm_worm.mat` and  (which uses `scipy.io.loadmat`)
- `example_video_feature_file.mat`,   (which uses `h5py`)

But it is giving an error when using:

- `example_contour_and_skeleton_info.mat` (which uses `h5py`)

For instance, here is the [error in build 192](https://travis-ci.org/openworm/movement_validation/builds/69826442):
```
======================================================================
ERROR: test_areas.test_areas
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda3/lib/python3.3/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/openworm/movement_validation/tests/test_areas.py", line 78, in test_areas
    bw = mv.BasicWorm.from_schafer_file_factory(schafer_bw_file_path)
  File "../movement_validation/prefeatures/basic_worm.py", line 208, in from_schafer_file_factory
    with h5py.File(data_file_path, 'r') as h:
  File "/home/travis/miniconda3/lib/python3.3/site-packages/h5py/_hl/files.py", line 260, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/home/travis/miniconda3/lib/python3.3/site-packages/h5py/_hl/files.py", line 89, in make_fid
    fid = h5f.open(name, flags, fapl=fapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper (-------src-dir-------/h5py/_objects.c:2458)
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper (-------src-dir-------/h5py/_objects.c:2415)
  File "h5py/h5f.pyx", line 76, in h5py.h5f.open (-------src-dir-------/h5py/h5f.c:1792)
OSError: Unable to open file (File signature not found)
```

The `test_areas.py` module works fine locally on my machine.  So it's something in the Travis CI configuration that's causing this to fail.

**Comment by MichaelCurrie** [comment]
It might be something to do with Google Drive file link.

**Comment by MichaelCurrie** [comment]
Ah, it appears that if I type the example_contour_and_skeleton_info.mat file into a browser, the file exceeds the maximum size Google can scan for viruses so there is an extra prompt before the download can occur:

![image](https://cloud.githubusercontent.com/assets/5267078/8554335/95a33b20-24a6-11e5-8323-ee3f63e45b15.png)

So there is some flag we need to add to `wget` I bet.

**Comment by MichaelCurrie** [comment]
This is a stackoverflow issue: http://stackoverflow.com/questions/20665881/direct-download-from-google-drive-using-google-drive-api

The base example_data folder is:

`https://drive.google.com/drive/u/0/folders/0B7to9gBdZEyGNWtWUElWVzVxc0E`

This tells us the UUID is `0B7to9gBdZEyGNWtWUElWVzVxc0E`.

We are supposed to treat this like a web host by prepending `https://googledrive.com/host/` to the the folder UUID using this URL:

`https://googledrive.com/host/0B7to9gBdZEyGNWtWUElWVzVxc0E/[file-name]`

So the URL needs to be:

`https://googledrive.com/host/0B7to9gBdZEyGNWtWUElWVzVxc0E/example_contour_and_skeleton_info.mat`

This works for me as a direct download; the antivirus prompt doesn't appear.  I'll change the `travis.yml` file and cross my fingers.

**Comment by MichaelCurrie** [comment]
Oops, accidentally closed this issue.

**Comment by JimHokanson** [comment]
The new approach requires a sign in ...

**Comment by JimHokanson** [comment]
Changed the permissions, should work now ...

**Comment by MichaelCurrie** [comment]
OK I just made an unrelated commit, that should trigger another Travis CI build, we shall see if it works.

**Comment by JimHokanson** [comment]
It is fine now. A summary:

1) Apparently the googledrive.com url is the best approach for downloading these files. Apparently over a certain file size limit the old way was prompting to download since a virus scan couldn't be run.
2) The permissions were also incorrectly set on the file. This might have been my fault from an earlier change trying to fix problem 1. I changed the settings again so that basically the file is open to viewing by the public.
3) A commit by @MichaelCurrie regarding test_areas had a bug when running on Travis.ci related to matplotlib. I added some instructions to the setup and it seems to be fixed now.
